src/common/fiscal-year.test.ts

```diff
@@ -1,25 +1,136 @@
+  start: Oct2020,
+  end: Sep302021,
+});
+
+/*-----------------------------------------------------------
+describe('fiscal-year-example', () => {
   //TODO - This is an example as to what a single test case WOULD be like with a
   // standard 'it' statement.  However, due to the nature of this function it should also be
   // transformed into the case below using the 'it.each' to easily cover all cases
   it('should return current year when date is in second quarter', () => {
-    const regularDate = DateTime.fromObject({ year: 2020, month: 6, day: 1 });
-
+    const regularDate = Jun2020;
     expect(fiscalYear(regularDate)).toEqual(2020);
   });
 });
+-------------------------------------------------------------*/
+
+describe('fiscalYear', () => {
+  it.each([
+    [Oct2019, 2020],
+    [Nov2019, 2020],
+    [Dec2019, 2020],
+    [Jan2020, 2020],
+    [Jun2020, 2020],
+  ])('FY%s -> CY%s', (fYear, cYear) => {
+    expect(fiscalYear(fYear)).toEqual(cYear);
+  });
+});
+
+describe('fiscalYears', () => {
+  it.each([
+    [Oct2019, Sep2020, [2020]],
+    [Oct2019, Oct2020, [2020, 2021]],
+    [Oct2019, undefined, []],
+    [undefined, Oct2020, []],
+    [Feb2019, Oct2020, [2019, 2020, 2021]],
+    [Jun2020, Feb2019, []],
+  ])('CStart%s CEnd%s -> expectedRange%s', (cStart, cEnd, expectedRange) => {
+    expect(fiscalYears(cStart, cEnd)).toEqual(expectedRange);
+  });
+});
+
+describe('fiscalQuarter', () => {
+  it.each([
+    [Oct2019, 1],
+    [Nov2019, 1],
+    [Dec2019, 1],
+    [Jan2020, 2],
+    [Jun2020, 3],
+    [Sep2020, 4],
+  ])('CalDate%s -> FiscalQtr%s', (cDate, expQtr) => {
+    expect(fiscalQuarter(cDate)).toEqual(expQtr);
+  });
+});
 
+describe('startOfFiscalYear', () => {
+  it.each([
+    [Oct2019, Oct2019],
+    [Nov2019, Oct2019],
+    [Dec2019, Oct2019],
+    [Jan2020, Oct2019],
+    [Sep2020, Oct2019],
+    [Oct2020, Oct2020],
+  ])('CalDate%s -> FiscalYr%s', (cDate, expYear) => {
+    expect(startOfFiscalYear(cDate)).toEqual(expYear);
+  });
+});
+
+describe('endOfFiscalYear', () => {
+  it.each([
+    [Feb2019, Sep302019],
+    [Oct2019, Sep302020],
+    [Nov2019, Sep302020],
+    [Dec2019, Sep302020],
+    [Jan2020, Sep302020],
+    [Oct2020, Sep302021],
+  ])('CalDate%s -> FiscalYr%s', (cDate, expYear) => {
+    expect(endOfFiscalYear(cDate)).toEqual(expYear);
+  });
+});
+
+describe('expandToFullFiscalYears', () => {
+  it.each([
+    [dInterval2020, dInterval2020],
+    [dInterval2021, dInterval2021],
+  ])('DateInterval%s -> FullFiscalYrs%s', (dateInt, expYears) => {
+    expect(expandToFullFiscalYears(dateInt)).toEqual(expYears);
+  });
+});
+
+describe('fullFiscalYear', () => {
+  it.each([
+    [2020, dInterval2020],
+    [2021, dInterval2021],
+  ])('YrNum%s -> FullFiscalYrInterval%s', (yrNum, fYearInterval) => {
+    expect(fullFiscalYear(yrNum)).toEqual(fYearInterval);
+  });
+});
 describe('fullFiscalQuarter', () => {
   it.each([
     [1, 2020, '2019-10-01/2019-12-31'],
```

Comment:
You can go ahead and make these constants like you did everything else.

Reply:
Sorry, I don't think I was clear here.  I'd propose making these strings into DateInterval constants up above.  For example,

```
const firstQuarter2021 = DateInterval.fromObject({
  start: Oct2019,
  end: Dec2019,
});
```

and then use it like this:

```
  it.each([
    [1, 2020, firstQuarter2021],
    ...
```

This would of course change the test to check for an Interval instead of a string, but it seems more consistent with what is done throughout the rest of the file now.

It doesn't need to happen though; just a suggestion.  If @CarsonF doesn't agree that's fine.

Reply:
I'd be fine with that. But "fiscal" should be in the name: `fiscalFiscalQuarter2021`

Reply:
I had to use DateInterval in a couple of the other functions because of the Arg Type definition, which is why I built some of those DateInterval consts, but your way Bryan on that particular function does not require a DateInterval parameter so its less lines of code and easier to leave the way you built it, but if you want me to still change it, I can revise. It will add more lines of code to create each Qtr range, let me know if you think it's worth it.

Reply:
You already have `qtrOne2020` and `qtrFour2020` defined, so you'd only have to add two more quarters, but whatever you want to do is cool.  I'm fine either way.

Reply:
updated with two more consts for qtrs three and four